pageserver/src/remote_storage/storage_sync.rs

```diff
@@ -1192,6 +1194,22 @@ fn register_sync_status(sync_start: Instant, sync_name: &str, sync_status: Optio
     .observe(secs_elapsed)
 }
 
+pub fn path_with_suffix_extension(original_path: impl AsRef<Path>, suffix: &str) -> PathBuf {
+    assert!(
+        !suffix.starts_with('.'),
+        "expected suffix without leading dot, got: {suffix}"
+    );
+    let new_extension = match original_path
+        .as_ref()
+        .extension()
+        .map(OsStr::to_string_lossy)
+    {
+        Some(extension) => format!("{extension}.{suffix}"),
+        None => suffix.to_owned(),
+    };
+    original_path.as_ref().with_extension(new_extension)
```

Comment:
NITs:

* you can use `Cow` and avoid cloning with `to_owned`: 
```rust
{
        Some(extension) => Cow::Owned(format!("{extension}.{suffix}")),
        None => Cow::Borrowed(suffix),
    };
    original_path.as_ref().with_extension(new_extension.as_ref())
```

* not super fond of this panic, might it be ok to be lenient and "merge" such dots (could be multiple dots though..)? Or even ignore them, let there be multiple dots in the name?

* might be good to add a negative test in case you want to fight dots